### PR TITLE
Print input/output values for each dispatch function in DEBUG mode

### PR DESCRIPTION
**Pull request by hanhanW**
Fixes https://github.com/google/iree/issues/2305

### CODE REVIEW

File: iree/compiler/Dialect/Flow/IR/FlowOps.td [hunk 1]
```diff
@@ -610,6 +610,17 @@ def FLOW_TensorUpdateOp : FLOW_PureOp<"tensor.update", [
   let hasFolder = 1;
 }
 
+def FLOW_TensorTraceOp : FLOW_Op<"tensor.trace", []> {
+  let summary = [{trace value(s) operation}];
+  let description = [{
+    Trace point for dispatchable functions.
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$operands);
```
Comment (benvanik) [inline iree/compiler/Dialect/Flow/IR/FlowOps.td:619]: you can change this to `Variadic<FLOW_Tensor>`

Comment (hanhanW) [reply iree/compiler/Dialect/Flow/IR/FlowOps.td]: Done

File: iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp [hunk 2]
```diff
@@ -57,11 +57,19 @@ LogicalResult convertToDispatchOp(DispatchRegionOp regionOp,
     return failure();
   }
 
+#ifndef NDEBUG
```
[the remark this answers is not in the document]
Comment (hanhanW) [reply iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: Done. Yeah, it's easier to use if make it a runtime flag.

File: iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp [hunk 3]
```diff
@@ -57,11 +57,19 @@ LogicalResult convertToDispatchOp(DispatchRegionOp regionOp,
...
+  builder.create<TensorTraceOp>(regionOp.getLoc(), newArgs);
```
[the remark this answers is not in the document]
Comment (hanhanW) [reply iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp]: Done

File: iree/compiler/Dialect/HAL/Conversion/FlowToHAL/ConvertTensorOps.cpp [hunk 4]
```diff
@@ -120,13 +121,40 @@ class TensorStoreOpConversion
...
+  TensorTraceOpConversion(MLIRContext *ctx, TypeConverter &converter)
+      : OpConversionPattern(ctx) {}
+
+  LogicalResult matchAndRewrite(
+      IREE::Flow::TensorTraceOp traceOp, llvm::ArrayRef<Value> rawOperands,
+      ConversionPatternRewriter &rewriter) const override {
+    Location loc = traceOp.getLoc();
+    SmallVector<Value, 4> bufferViews;
+    for (auto operand : llvm::enumerate(rawOperands)) {
+      if (!IREE::HAL::TensorRewriteAdaptor::isValidNewType(
```
[the remark this answers is not in the document]
Comment (hanhanW) [reply iree/compiler/Dialect/HAL/Conversion/FlowToHAL/ConvertTensorOps.cpp]: Done

File: iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertBufferViewOps.cpp [hunk 5]
```diff
@@ -100,6 +100,8 @@ void populateHALBufferViewToVMPatterns(MLIRContext *context,
                                        SymbolTable &importSymbols,
                                        TypeConverter &typeConverter,
                                        OwningRewritePatternList &patterns) {
+  patterns.insert<VMImportOpConversion<IREE::HAL::BufferViewTraceOp>>(
```
[the remark this answers is not in the document]
Comment (hanhanW) [reply iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertBufferViewOps.cpp]: Oh, missed this during cleanup. Done.

**Pull request merged**